### PR TITLE
Support customizing the UI

### PR DESCRIPTION
**Pull request by AbdulazizAlrabiah**

### CODE REVIEW

File: lib/src/moyasar.dart [hunk 1]
```diff
@@ -7,6 +7,9 @@ import 'package:moyasar/src/models/payment_request.dart';
 
 const version = "2.0.3";
```
Comment (Fghurayri) [inline lib/src/moyasar.dart:8]: Please bump this version number as well

File: lib/moyasar.dart [hunk 2]
```diff
@@ -1,18 +1,29 @@
 /// Moyasar Flutter SDK helps you accept payments quickly and securely.
 library moyasar;
 
+export 'src/moyasar.dart' show Moyasar;
+
 export 'src/widgets/credit_card.dart' show CreditCard;
 export 'src/widgets/apple_pay.dart' show ApplePay;
+export 'src/widgets/three_d_s_webview.dart' show ThreeDSWebView;
```
Comment (Fghurayri) [inline lib/moyasar.dart:8]: Since the goal is to support custom UI, then no need to export this internal widget

File: lib/moyasar.dart [hunk 3]
```diff
@@ -1,18 +1,29 @@
...
+export 'src/widgets/network_icons.dart' show NetworkIcons;
```
Comment (Fghurayri) [inline lib/moyasar.dart:9]: Same as above, no need to export this internal widget

**Review by Fghurayri — CHANGES_REQUESTED** [review]
Hey @AbdulazizAlrabiah 👋

Thanks for your awesome job 👏 This is a highly-requested feature by our customers

I left a few comments for you. Moreover, here are a few more that are outside the scope of changed files:

- Let's satisfy Dart's code analyzer which helps our Pub Dev scores. So, please utilize your editor to track down and Quick-Fix all newly introduced problems (like unused or unnecessary imports)

- Let's get an updated `pubspec.lock` file under the `/example` directory which helps keeping the integrity of all dependencies versions. So, please run `flutter pub get` in the root directory and get the updated file committed to this PR as well.

- Let's update our Readme to include some information about this new feature. I suggest adding two sub-headers under `Usage` to provide two options:

  1. Moyasar Widgets (this is already outlined, just add this sub-header above the current code sample)
  2. Custom Widgets

Add this sample code to showcase using the Moyasar API with custom widgets:

```dart
  // Unified config for Moyasar API
  final paymentConfig = PaymentConfig(
    publishableApiKey: 'YOUR_API_KEY',
    amount: 25758, // SAR 257.58
    description: 'order #1324',
    metadata: {'size': '250g'},
    creditCard: CreditCardConfig(saveCard: true, manual: false),
    applePay: ApplePayConfig(
        merchantId: 'YOUR_MERCHANT_ID',
        label: 'YOUR_STORE_NAME',
        manual: false),
  );

  // Callback once the user clicks on the custom Apple Pay widget
  void onSubmitApplePay(applePay) async {
    final source = ApplePayPaymentRequestSource(
        applePay['token'], (paymentConfig.creditCard as ApplePayConfig).manual);
    final paymentRequest = PaymentRequest(paymentConfig, source);

    final result = await Moyasar.pay(
        apiKey: paymentConfig.publishableApiKey,
        paymentRequest: paymentRequest);

    onPaymentResult(result);
  }

  // Callback once the user fills & submit their CC information using the custom form widget
  void onSubmitCcForm() async {
    final source = CardPaymentRequestSource(
        creditCardData: CardFormModel(
            name: 'John Doe',
            number: '4111111111111111',
            month: '05',
            year: '2025'),
        tokenizeCard: (paymentConfig.creditCard as CreditCardConfig).saveCard,
        manualPayment: (paymentConfig.creditCard as CreditCardConfig).manual);

    final paymentRequest = PaymentRequest(paymentConfig, source);

    final result = await Moyasar.pay(
        apiKey: paymentConfig.publishableApiKey,
        paymentRequest: paymentRequest);

    onPaymentResult(result);
  }

  // Unified payment result processor
  void onPaymentResult(result) {
    if (result is PaymentResponse) {
      switch (result.status) {
        case PaymentStatus.initiated:
          // handle 3DS redirection.
          break;
        case PaymentStatus.paid:
          // handle success.
          break;
        case PaymentStatus.failed:
          // handle failure.
          break;
      }
    }
  }
```

Please let me know if you need any help or clarifications.

File: lib/moyasar.dart (hunk 2 above)
Comment (AbdulazizAlrabiah) [reply lib/moyasar.dart:8]: I thought it will be easier if developers needed it, but sure I'll remove it

**Comment by AbdulazizAlrabiah** [comment]
Hello @Fghurayri 👋

Thank you for the great review and suggestion!🙏

Working on it 👍

**Comment by AbdulazizAlrabiah** [comment]
PR is ready 😙

**Comment by Fghurayri** [comment]
Great job 👏 @ahoshaiyan Please go ahead and merge this PR and I'll publish a new version once that's done 🚀

**Pull request merged**